doc/docs/master/how-tos/enhanced-pipeline-workflow.md

```diff
@@ -0,0 +1,121 @@
+
+Build pipelines require a modification to the pipeline spec's transform object with options for the following fields:
+
+- path: An optional string specifying where the source code is relative to the pipeline spec path (or the cwd if the pipeline is fed into pachctl via stdin.)
+- language: A string specifying what language builder to use (see below.) Only works with official builders.
+- image: A string specifying what builder image to use, if a non-official builder is desired.
+
+Below is a Python example of a build pipline.
+
+```json
+{
+  "pipeline": {
+    "name": "map"
+  },
+  "description": "A pipeline that tokenizes scraped pages and appends counts of words to corresponding files.",
+  "transform": {
+    "build": {
+      "language": "python",
+      "path": "./source"
+    }
+  },
+  "input": {
+    "pfs": {
+      "repo": "scraper",
+      "glob": "/*"
+    }
+  }
+}
+```
+
+A build pipeline can be submitted the same way as any other pipeline, for example:
+
+```bash
+pachctl update pipeline -f <pipeline name> 
+```
+
+When submitted, the following actions occur: 
+
+1. All files (code, etc.) are copied from the build path to a PFS repository, `<pipeline name>_build`, which we can think of as the source code repository. In the case above, everything in `./source` would be copied to to the PFS `<pipeline name>_build` repository.
+
+1. Starts a pipeline called `<pipeline name>_build` that reads the files uploaded and runs the `build.sh` script to pull dependencies and compile any requirements from the source code. 
+
+1. Creates (or updates if it already exists) the running pipeline `<pipeline name>` with the the newly creates source files and built assets. 
+
+!!! note
+      You can optionally specify a `.pachignore` file to prevent certain files from getting pushed to this repo. 
+
+The updated pipeline contains the following PFS repos mapped in as inputs:
+
+1. `/pfs/source` - source code that is required for running the pipeline. There are language-specific requirements on the structure of this source code mentioned in the [Python Builder](#python-builder) section.
+
+1. `/pfs/build` - any artifacts resulting from the build process. 
+
+1. `/pfs/<input(s)>` - any inputs specified in the pipeline specification
+
+### Builders
+Either `language` or `image` must be specified in the pipeline spec. The langauge parameter informs the pipeline of what base image to use and already contain implementations for `build.sh` and `run.sh`. Go and Python "builders" have a standarized implementation, only requiring setting the `language` parameter in the pipeline specification.  
+
+If `language` is not specified, then the build pipeline uses the `transform.image` container (if `transform.image` is not specified the builder image is used). Similarly, if `transform.cmd` is not specified, it will default to running `sh /pfs/build/run.sh`, which should be provided via the build pipeline.
```

Comment:
This is slightly incorrect, as it follows this precedence order:

* `transform.build.language`
* `transform.build.image`
* `transform.image`

Reply:
Updated in most recent commit.